muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/chat/ChatSyncController.java

```diff
@@ -0,0 +1,218 @@
+        byte[] passwordBytes = new byte[20];
+        random.nextBytes(passwordBytes);
+        String password = Base64.encodeBase64String(passwordBytes);
+
+        userEntity = new UserEntity(studentIdentifier.toId(), user.getDisplayName(), "", password);
+        client.createUser(userEntity);
+
+        if (userSchoolDataSource == null || userIdentifier == null) {
+          logger.log(Level.WARNING, "No user entity found for identifier " + studentIdentifier.getIdentifier() + ", skipping...");
+        }
+      }
+        
+      List<WorkspaceEntity> usersWorkspaces = workspaceController.listWorkspaceEntitiesByUser(userEntityController.findUserEntityByUser(user), true);
+
+      for (WorkspaceEntity usersWorkspace : usersWorkspaces) {
+        MUCRoomEntity chatRoomEntity = client.getChatRoom(usersWorkspace.getIdentifier());
+        Workspace workspace1 = workspaceController.findWorkspace(usersWorkspace);
+        Set<SchoolDataIdentifier> curriculumIdentifiers = workspace1.getCurriculumIdentifiers();
+        boolean hasCorrectCurriculums = true;
+          
+        for (SchoolDataIdentifier curriculumIdentifier : curriculumIdentifiers) {
+          curriculumIdentifier.getIdentifier();
+          
+          Curriculum curriculum = courseMetaController.findCurriculum(curriculumIdentifier);
+            
+          String curriculumName = curriculum.getName();
+            
+          if (curriculumName.equals("OPS2005")) {
+            hasCorrectCurriculums = false;
+            break;
+          } 
+             
+        }
+            
+        if (hasCorrectCurriculums) {
+          if (chatRoomEntity == null) {
+            logger.log(Level.INFO, "Syncing chat workspace " + usersWorkspace.getUrlName());
+            if (identifier == null) {
+              logger.log(Level.WARNING, "Invalid workspace identifier " + identifier + ", skipping...");
+              continue;
+            }
+            
+            Workspace workspace = workspaceController.findWorkspace(usersWorkspace);
+            String subjectCode = courseMetaController.findSubject(workspace.getSchoolDataSource(), workspace.getSubjectIdentifier()).getCode();
+              
+            String roomName = subjectCode + workspace.getCourseNumber() + " - " + workspace.getNameExtension();
+              
+            chatRoomEntity = new MUCRoomEntity(workspace.getIdentifier(), roomName, workspace.getDescription());
+            chatRoomEntity.setPersistent(true);
+            chatRoomEntity.setLogEnabled(true);
+            chatRoomEntity.setPublicRoom(true);
+            client.createChatRoom(chatRoomEntity);
+              
+            EnvironmentUser workspaceUserRole = environmentUserController.findEnvironmentUserByUserEntity(userEntityController.findUserEntityByUser(user)); 
+            EnvironmentRoleEntity role = workspaceUserRole.getRole();
+            if (role.toString().equals("ADMINISTRATOR") || role.toString().equals("PROGRAMME-STUDY-LEADER")) {
```

Comment:
EnvironmentRoleEntity.toString() probably doesn't return what is asked here. Should it be  EnvironmentRoleArchetype.ADMINISTRATOR.equals(role.getArchetype()) (and the same for the other role?